llog/levellog.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// llog is a dirt-simple leveled text logger.
+package llog
+
+import (
+	"log"
+	"log/slog"
+	"math"
+	"testing"
+)
+
+// Log is the stdlib default log sink.
+var Default *Logger = &Logger{Sink: log.Printf}
+
+// Null is a logger that prints nothing.
+var Null *Logger = &Logger{Sink: nil}
+
+// Test is a logger that prints every level to t.Logf.
+func Test(t testing.TB) *Logger {
+	return &Logger{Sink: t.Logf, Level: slog.Level(math.MinInt32)}
+}
+
+// Debug prints to log.Printf at the debug level.
+var Debug *Logger = &Logger{Sink: log.Printf, Level: slog.LevelDebug}
+
+// Printf is a logger printf function.
+type Printf func(format string, v ...any)
+
+// Logger is a dirt-simple leveled logger.
+//
+// If the log level is >= Level, it logs to the given Sink.
+//
+// Logger and Sink may be nil in order to log nothing.
+type Logger struct {
```

Comment:
I like this, it's simple and it looks like it would work.
I wonder about  making Sink and Level private and creating a New function. Otherwise, you're going to find people doing stuff like this:
llog.Default.Printf = bla bla
which I'm pretty sure is going to be not what we want. 
Also if you provde a (l*logger)SetLevel(level) error
then you have a way for people to learn about trying to set a bogus level ...

Also, in line with the whole "no global" thing, should Default and Null be func?

Reply:
I went for something in the middle. PTAL.